iommu_sw_guidelines.adoc

```diff
@@ -53,18 +53,19 @@ The guidelines for initializing the IOMMU are as follows:
   `msi_data_V` register with value `D`. Configure the interrupt mask `M` in
   `msi_vec_ctl_V` register appropriately.
 . To program the command queue, first determine the number of entries `N` needed
-  in the command queue. The number of entries in the command queue is always a
+  in the command queue. The number of entries in the command queue must be a
   power of two. If `N` is 256 or lower then allocate a memory buffer that is
   aligned to a 4-KiB page address and is of size `N` x 16-bytes.  If `N` is
   greater than 256 then allocate a memory buffer that is naturally aligned to a
-  `N` x 16-byte address boundary. Let `k=log~2~(N)` and `B` be the physical page
-  number (PPN) of the allocated memory buffer. Program the command queue
-  registers as follows:
+  `N` x 16-byte address boundary and is of size `N` x 16-bytes. Let `k=log~2~(N)`
```

Comment:
Don't we usually say that it must be size aligned ?

Reply:
Updated as discussed. Please see the new text.